InnerEye/ML/metrics.py

```diff
@@ -91,180 +321,100 @@ def log_metrics(self, run_context: Run = None) -> None:
         :return:
         """
         run_context = get_run_context_or_default(run_context)
-        keys = sorted(self.epochs.keys())
 
         run_context.log_table(name=self.get_metrics_log_key(), value={
-            "Checkpoint": keys,
-            "Dice": [self.epochs[x] for x in keys]
+            "Dice": self.metrics
         })
 
 
-@dataclass(frozen=True)
-class SegmentationMetricsPerClass:
+@dataclass
+class EpochTimers:
     """
-    Stores different segmentation metrics, as a list where each list entry represents one class.
+    Contains all information necessary to compute the IO metrics: Epoch times, batch times, loading times.
     """
-    dice: List[float] = field(default_factory=list)
-    hausdorff_distance_mm: List[float] = field(default_factory=list)
-    mean_distance_mm: List[float] = field(default_factory=list)
+    epoch_start_time: float = time.time()
+    epoch_end_time: float = time.time()
+    batch_start_time: float = time.time()
+    num_load_time_warnings: int = 0
+    num_load_time_exceeded: int = 0
+    total_extra_load_time: float = 0.0
+    total_load_time: float = 0.0
+    num_batches: int = 0
+    load_time_warning_epochs: Set[int] = field(default_factory=set)
 
-    def append_nan(self) -> None:
+    def reset(self) -> None:
         """
-        Adds a NaN for all metrics that are stored, indicating that the target class was not present and no
-        output was produced by the model.
+        Resets all timers to the current time, and all counters to 0. The set of epochs for which warnings about
+        load time were produced will not be reset.
         """
-        self.dice.append(math.nan)
-        self.hausdorff_distance_mm.append(math.nan)
-        self.mean_distance_mm.append(math.nan)
-
-    def append(self,
-               dice: float,
-               hausdorff_distance_mm: float,
-               mean_distance_mm: float) -> None:
+        current_time = time.time()
```

Comment:
Not sure how much it matters for the typical orders of magnitude here, but could it not be better to use [`time.perf_counter()`](https://docs.python.org/3/library/time.html#time.perf_counter), which uses the highest resolution timer available, instead of `time.time()`?

Reply:
We are logging metrics at seconds resolution, so I don't think this really matters here.